### ISSUE TITLE
Pushing multi arch manifest to quay.io fails with 'invalid manifest'

### ISSUE THREAD

**Issue by jadahl** [containers/buildah#2724]
**Description**

Attempting to push a multi arch manifest (using one ARM64 and one AMD64 image) fails with

```
Uploading manifest list failed, attempted the following formats: application/vnd.oci.image.index.v1+json(Error uploading manifest buildah-2020-10-27.1 to quay.io/jadahl/ci-templates: manifest invalid: manifest invalid), application/vnd.docker.distribution.manifest.list.v2+json(Error uploading manifest buildah-2020-10-27.1 to quay.io/jadahl/ci-templates: manifest invalid: manifest invalid)
```

It sounds strikingly similar to https://github.com/containers/buildah/issues/2594 but with the `--format` work around it still doesn't work. I also tried using buildah from git (b8a4ae628da277f860aa59d13fae890fe48339d0), and the result is the same.

Pushing to e.g. `registry.freedesktop.org` works, and fetching the resulting image will result in the right arch depending on how/from where it's fetched.

The debug log for the unsuccessful push is avalable here: 
[manifest-push-fails.txt](https://github.com/containers/buildah/files/5444414/manifest-push-fails.txt)

**Steps to reproduce the issue:**

I did the following, and it falis at the last step (`buildah manifest push`).

```sh
buildah manifest create buildah-2020-10-27.1
buildah manifest add --override-arch=amd64 --override-os=linux --arch=amd64 buildah-2020-10-27.1 docker://registry.freedesktop.org/jadahl/ci-templates/buildah:2020-10-26.3
buildah manifest add --override-arch=arm64 --override-os=linux --arch=arm64 --variant=v8 buildah-2020-10-27.1 docker://registry.freedesktop.org/jadahl/ci-templates/arm64v8/buildah:2020-10-26.3
buildah manifest push --all buildah-2020-10-27.1 docker://quay.io/jadahl/ci-templates
```

The image `docker://registry.freedesktop.org/jadahl/ci-templates/buildah:2020-10-26.3` contains an amd64 image.
The image `docker://registry.freedesktop.org/jadahl/ci-templates/arm64v8/buildah:2020-10-26.3` contains an arm64v8 image.

Replacing the last line with

```sh
buildah manifest push --all buildah-2020-10-27.1 docker://registry.freedesktop.org/jadahl/ci-templates:buildah-2020-10-27.1
```

succeeds.

**Describe the results you received:**

Instead of successfully pushing the manifest, I receive an error (see description).

**Describe the results you expected:**

I expected it to push and later be accessible for downloading, as happens with other registries.

**Output of `rpm -q buildah` or `apt list buildah`:**

```
buildah-1.16.2-1.fc33.x86_64
```
though I have also `make install`:ed from git, without any improvements.

**Output of `buildah version`:**

```
buildah version 1.17.0-dev (image-spec 1.0.1-dev, runtime-spec 1.0.2-dev)
```

**Output of `podman version` if reporting a `podman build` issue:**

```
podman version 2.1.0-dev
```

**Output of `cat /etc/*release`:**

```
Fedora release 33 (Thirty Three)
NAME=Fedora
VERSION="33 (Workstation Edition Prerelease)"
ID=fedora
VERSION_ID=33
VERSION_CODENAME=""
PLATFORM_ID="platform:f33"
PRETTY_NAME="Fedora 33 (Workstation Edition Prerelease)"
ANSI_COLOR="0;38;2;60;110;180"
LOGO=fedora-logo-icon
CPE_NAME="cpe:/o:fedoraproject:fedora:33"
HOME_URL="https://fedoraproject.org/"
DOCUMENTATION_URL="https://docs.fedoraproject.org/en-US/fedora/f33/system-administrators-guide/"
SUPPORT_URL="https://fedoraproject.org/wiki/Communicating_and_getting_help"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=33
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=33
PRIVACY_POLICY_URL="https://fedoraproject.org/wiki/Legal:PrivacyPolicy"
VARIANT="Workstation Edition"
VARIANT_ID=workstation
Fedora release 33 (Thirty Three)
Fedora release 33 (Thirty Three)
```

**Output of `uname -a`:**

```
Linux shuben 5.8.6-301.fc33.x86_64 #1 SMP Fri Sep 4 04:57:36 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
```

**Output of `cat /etc/containers/storage.conf`:**

```
# This file is is the configuration file for all tools
# that use the containers/storage library.
# See man 5 containers-storage.conf for more information
# The "container storage" table contains all of the server options.
[storage]

# Default Storage Driver
driver = "overlay"

# Temporary storage location
runroot = "/var/run/containers/storage"

# Primary Read/Write location of container storage
graphroot = "/var/lib/containers/storage"

# Storage path for rootless users
#
# rootless_storage_path = "$HOME/.local/share/containers/storage"

[storage.options]
# Storage options to be passed to underlying storage drivers

# AdditionalImageStores is used to pass paths to additional Read/Only image stores
# Must be comma separated list.
additionalimagestores = [
]

# Remap-UIDs/GIDs is the mapping from UIDs/GIDs as they should appear inside of
# a container, to the UIDs/GIDs as they should appear outside of the container,
# and the length of the range of UIDs/GIDs.  Additional mapped sets can be
# listed and will be heeded by libraries, but there are limits to the number of
# mappings which the kernel will allow when you later attempt to run a
# container.
#
# remap-uids = 0:1668442479:65536
# remap-gids = 0:1668442479:65536

# Remap-User/Group is a user name which can be used to look up one or more UID/GID
# ranges in the /etc/subuid or /etc/subgid file.  Mappings are set up starting
# with an in-container ID of 0 and then a host-level ID taken from the lowest
# range that matches the specified name, and using the length of that range.
# Additional ranges are then assigned, using the ranges which specify the
# lowest host-level IDs first, to the lowest not-yet-mapped in-container ID,
# until all of the entries have been used for maps.
#
# remap-user = "containers"
# remap-group = "containers"

# Root-auto-userns-user is a user name which can be used to look up one or more UID/GID
# ranges in the /etc/subuid and /etc/subgid file.  These ranges will be partitioned
# to containers configured to create automatically a user namespace.  Containers
# configured to automatically create a user namespace can still overlap with containers
# having an explicit mapping set.
# This setting is ignored when running as rootless.
# root-auto-userns-user = "storage"
#
# Auto-userns-min-size is the minimum size for a user namespace created automatically.
# auto-userns-min-size=1024
#
# Auto-userns-max-size is the minimum size for a user namespace created automatically.
# auto-userns-max-size=65536

[storage.options.overlay]
# ignore_chown_errors can be set to allow a non privileged user running with
# a single UID within a user namespace to run containers. The user can pull
# and use any image even those with multiple uids.  Note multiple UIDs will be
# squashed down to the default uid in the container.  These images will have no
# separation between the users in the container. Only supported for the overlay
# and vfs drivers.
#ignore_chown_errors = "false"

# Path to an helper program to use for mounting the file system instead of mounting it
# directly.
#mount_program = "/usr/bin/fuse-overlayfs"

# mountopt specifies comma separated list of extra mount options
mountopt = "nodev,metacopy=on"

# Set to skip a PRIVATE bind mount on the storage home directory.
# skip_mount_home = "false"

# Size is used to set a maximum size of the container image.
# size = ""

[storage.options.thinpool]
# Storage Options for thinpool

# autoextend_percent determines the amount by which pool needs to be
# grown. This is specified in terms of % of pool size. So a value of 20 means
# that when threshold is hit, pool will be grown by 20% of existing
# pool size.
# autoextend_percent = "20"

# autoextend_threshold determines the pool extension threshold in terms
# of percentage of pool size. For example, if threshold is 60, that means when
# pool is 60% full, threshold has been hit.
# autoextend_threshold = "80"

# basesize specifies the size to use when creating the base device, which
# limits the size of images and containers.
# basesize = "10G"

# blocksize specifies a custom blocksize to use for the thin pool.
# blocksize="64k"

# directlvm_device specifies a custom block storage device to use for the
# thin pool. Required if you setup devicemapper.
# directlvm_device = ""

# directlvm_device_force wipes device even if device already has a filesystem.
# directlvm_device_force = "True"

# fs specifies the filesystem type to use for the base device.
# fs="xfs"

# log_level sets the log level of devicemapper.
# 0: LogLevelSuppress 0 (Default)
# 2: LogLevelFatal
# 3: LogLevelErr
# 4: LogLevelWarn
# 5: LogLevelNotice
# 6: LogLevelInfo
# 7: LogLevelDebug
# log_level = "7"

# min_free_space specifies the min free space percent in a thin pool require for
# new device creation to succeed. Valid values are from 0% - 99%.
# Value 0% disables
# min_free_space = "10%"

# mkfsarg specifies extra mkfs arguments to be used when creating the base
# device.
# mkfsarg = ""

# metadata_size is used to set the `pvcreate --metadatasize` options when
# creating thin devices. Default is 128k
# metadata_size = ""

# Size is used to set a maximum size of the container image.
# size = ""

# use_deferred_removal marks devicemapper block device for deferred removal.
# If the thinpool is in use when the driver attempts to remove it, the driver
# tells the kernel to remove it as soon as possible. Note this does not free
# up the disk space, use deferred deletion to fully remove the thinpool.
# use_deferred_removal = "True"

# use_deferred_deletion marks thinpool device for deferred deletion.
# If the device is busy when the driver attempts to delete it, the driver
# will attempt to delete device every 30 seconds until successful.
# If the program using the driver exits, the driver will continue attempting
# to cleanup the next time the driver is used. Deferred deletion permanently
# deletes the device and all data stored in device will be lost.
# use_deferred_deletion = "True"

# xfs_nospace_max_retries specifies the maximum number of retries XFS should
# attempt to complete IO when ENOSPC (no space) error is returned by
# underlying storage device.
# xfs_nospace_max_retries = "0"
```

**Comment by nalind** [comment]
If you remove `~/.local/share/containers/storage/cache/blob-info-cache-v1.boltdb` before attempting to push the list, does the push attempt succeed?

**Comment by jadahl** [comment]
> If you remove `~/.local/share/containers/storage/cache/blob-info-cache-v1.boltdb` before attempting to push the list, does the push attempt succeed?

It doesn't seem to have any affect, it still fails with the same error.

**Comment by mh21** [comment]
Something new I learned in https://github.com/containers/buildah/issues/2594 was that also `manifest push` has a `--format` parameter. Does adding `--format v2s2` there do something for you?

**Comment by jadahl** [comment]
I was aware of the (at that time closed) #2594 issue so I tried the --format work around but at that time it didn't make any difference.

**Comment by rhatdan** [comment]
I believe quay.io and buildah are now fixed.  Reopen if I am mistaken.

**Comment by dghubble** [comment]
Fixed on F33 latest (v1.18.0) or project latest (v1.19.4)? I rolled back a build pipeline from f33 v1.18.0 to buildah v1.16.2 to workaround.
